tasties_app/views.py

```diff
@@ -1,63 +1,84 @@
-from django.shortcuts import render, redirect
-from tasties_app.models import Recipe, Rating
-from django.db.models import Avg
-from collections import OrderedDict
-from django.contrib.auth import logout, login, authenticate
+from django.contrib import messages
+from django.contrib.auth import authenticate, login, logout
 from django.contrib.auth.decorators import login_required
+from django.db.models import Avg
+from django.shortcuts import redirect, render
+from tasties_app.models import Category, Recipe
+
 from .forms import CreateUserForm
-from django.contrib import messages
 
 
 def base(request):
-    return render(request, 'tasties_app/base.html',)
+    return render(
+        request,
+        "tasties_app/base.html",
+    )
 
 
-@login_required(login_url='login')
+@login_required(login_url="login")
 def recipes(request):
-    recipes_list = Recipe.objects.all()
-    recipes_with_ratings = {}
-    for recipe in recipes_list:
-        recipes_with_ratings[recipe] = Rating.objects.filter(recipe_id=recipe).aggregate(Avg('rating'))['rating__avg']
-    recipes_with_ratings = OrderedDict(sorted(recipes_with_ratings.items(), key=lambda x: x[1], reverse=True))
-    context = {'recipes_with_ratings': recipes_with_ratings}
-    return render(request, 'tasties_app/recipes.html', context)
+    recipes_list = Recipe.objects.annotate(
+        recipe_rating=Avg("rating__rating")
+    ).order_by("-recipe_rating")
+    categories_list = Category.objects.all()
+    selected_category = request.GET.get("category")
+    if selected_category:
+        if selected_category != "remove_filter":
```

Comment:
Here we can just say - if there is no "category" query param - don't filter.
Without the "remove_filter" option.

Reply:
we need to click on something in order to remove  the filters and displays all the recipes